src/db/gc/mod.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
```

Comment:
Module-level documentation is required here. Should include:

- What problem are we trying to solve? (I.e., why is there garbage? What _is_ garbage?)
- Overview of the solution space. What other solutions exist, and why did we choose this particular solution?
- List the special constraints and domain knowledge. What kind of data are we dealing with? What does it mean that the graph is directed? What does it mean that the data is persistent? What does it mean that there can be multiple, concurrent branches in the blockchain? How does `chain finality` affect the GC?
- Performance characteristics. Include both algorithm complexities and empirical measurements. I.e., how much space is the GC expected to use? How much time does it take to do a GC run with a lite node where all the data is in ParityDB?